### ISSUE TITLE
Documentation for gee-0.8 not present

### ISSUE THREAD

**Issue by astavale** [Valadoc/valadoc-org#45]
1. Using the direct link, http://valadoc.org/gee-0.8/index.htm , results in a blank page and sidebar menu.

2. Following the gee-0.8 link in the sidebar menu gives the error "Error 0: . When loading /gee-0.8/index."

3. Searching for a Gee data structure, for example TreeMap, shows no results

**Comment by btkostner** [comment]
I would guess that is due to an error while generating the docs, but I can confirm once I get back to a computer in a couple of days

**Comment by btkostner** [comment]
This seems like the relevent build logs:
```
Archive:  libgee-0.18.0.zip
4c5a09345a7e15180a7615a57fa23e42d89005b9
   creating: libgee-0.18.0/
  inflating: libgee-0.18.0/.gitignore  
  inflating: libgee-0.18.0/.mailmap  
 extracting: libgee-0.18.0/AUTHORS   
  inflating: libgee-0.18.0/COPYING   
  inflating: libgee-0.18.0/ChangeLog.pre-0-1-5  
  inflating: libgee-0.18.0/MAINTAINERS  
  inflating: libgee-0.18.0/Makefile.am  
  inflating: libgee-0.18.0/Makefile.decl  
  inflating: libgee-0.18.0/NEWS      
  inflating: libgee-0.18.0/README    
  inflating: libgee-0.18.0/autogen.sh  
   creating: libgee-0.18.0/benchmark/
  inflating: libgee-0.18.0/benchmark/Makefile.am  
  inflating: libgee-0.18.0/benchmark/benchmark.vala  
  inflating: libgee-0.18.0/benchmark/benchmarksorts.vala  
  inflating: libgee-0.18.0/benchmark/collections.vala  
  inflating: libgee-0.18.0/configure.ac  
   creating: libgee-0.18.0/doc/
  inflating: libgee-0.18.0/doc/Makefile.am  
  inflating: libgee-0.18.0/doc/index.wiki  
  inflating: libgee-0.18.0/gee-0.8.pc.in  
   creating: libgee-0.18.0/gee/
  inflating: libgee-0.18.0/gee/Makefile.am  
  inflating: libgee-0.18.0/gee/abstractbidirlist.vala  
  inflating: libgee-0.18.0/gee/abstractbidirsortedmap.vala  
  inflating: libgee-0.18.0/gee/abstractbidirsortedset.vala  
  inflating: libgee-0.18.0/gee/abstractcollection.vala  
  inflating: libgee-0.18.0/gee/abstractlist.vala  
  inflating: libgee-0.18.0/gee/abstractmap.vala  
  inflating: libgee-0.18.0/gee/abstractmultimap.vala  
  inflating: libgee-0.18.0/gee/abstractmultiset.vala  
  inflating: libgee-0.18.0/gee/abstractqueue.vala  
  inflating: libgee-0.18.0/gee/abstractset.vala  
  inflating: libgee-0.18.0/gee/abstractsortedmap.vala  
  inflating: libgee-0.18.0/gee/abstractsortedset.vala  
  inflating: libgee-0.18.0/gee/arraylist.vala  
  inflating: libgee-0.18.0/gee/arrayqueue.vala  
  inflating: libgee-0.18.0/gee/assemblyinfo.vala  
  inflating: libgee-0.18.0/gee/bidiriterator.vala  
  inflating: libgee-0.18.0/gee/bidirlist.vala  
  inflating: libgee-0.18.0/gee/bidirlistiterator.vala  
  inflating: libgee-0.18.0/gee/bidirmapiterator.vala  
  inflating: libgee-0.18.0/gee/bidirsortedmap.vala  
  inflating: libgee-0.18.0/gee/bidirsortedset.vala  
  inflating: libgee-0.18.0/gee/collection.vala  
  inflating: libgee-0.18.0/gee/comparable.vala  
  inflating: libgee-0.18.0/gee/concurrentlist.vala  
  inflating: libgee-0.18.0/gee/concurrentset.vala  
  inflating: libgee-0.18.0/gee/deque.vala  
  inflating: libgee-0.18.0/gee/functions.vala  
  inflating: libgee-0.18.0/gee/future.vala  
  inflating: libgee-0.18.0/gee/hashable.vala  
  inflating: libgee-0.18.0/gee/hashmap.vala  
  inflating: libgee-0.18.0/gee/hashmultimap.vala  
  inflating: libgee-0.18.0/gee/hashmultiset.vala  
  inflating: libgee-0.18.0/gee/hashset.vala  
  inflating: libgee-0.18.0/gee/hazardpointer.vala  
  inflating: libgee-0.18.0/gee/iterable.vala  
  inflating: libgee-0.18.0/gee/iterator.vala  
  inflating: libgee-0.18.0/gee/lazy.vala  
  inflating: libgee-0.18.0/gee/lightmapfuture.vala  
  inflating: libgee-0.18.0/gee/linkedlist.vala  
  inflating: libgee-0.18.0/gee/list.vala  
  inflating: libgee-0.18.0/gee/listiterator.vala  
  inflating: libgee-0.18.0/gee/map.vala  
  inflating: libgee-0.18.0/gee/mapiterator.vala  
  inflating: libgee-0.18.0/gee/multimap.vala  
  inflating: libgee-0.18.0/gee/multiset.vala  
  inflating: libgee-0.18.0/gee/priorityqueue.vala  
  inflating: libgee-0.18.0/gee/promise.vala  
  inflating: libgee-0.18.0/gee/queue.vala  
  inflating: libgee-0.18.0/gee/readonlybidirlist.vala  
  inflating: libgee-0.18.0/gee/readonlybidirsortedmap.vala  
  inflating: libgee-0.18.0/gee/readonlybidirsortedset.vala  
  inflating: libgee-0.18.0/gee/readonlycollection.vala  
  inflating: libgee-0.18.0/gee/readonlylist.vala  
  inflating: libgee-0.18.0/gee/readonlymap.vala  
  inflating: libgee-0.18.0/gee/readonlymultimap.vala  
  inflating: libgee-0.18.0/gee/readonlymultiset.vala  
  inflating: libgee-0.18.0/gee/readonlyset.vala  
  inflating: libgee-0.18.0/gee/readonlysortedmap.vala  
  inflating: libgee-0.18.0/gee/readonlysortedset.vala  
  inflating: libgee-0.18.0/gee/set.vala  
  inflating: libgee-0.18.0/gee/sortedmap.vala  
  inflating: libgee-0.18.0/gee/sortedset.vala  
  inflating: libgee-0.18.0/gee/streamiterator.vala  
  inflating: libgee-0.18.0/gee/task.vala  
  inflating: libgee-0.18.0/gee/teeiterator.vala  
  inflating: libgee-0.18.0/gee/timsort.vala  
  inflating: libgee-0.18.0/gee/traversable.vala  
  inflating: libgee-0.18.0/gee/treemap.vala  
  inflating: libgee-0.18.0/gee/treemultimap.vala  
  inflating: libgee-0.18.0/gee/treemultiset.vala  
  inflating: libgee-0.18.0/gee/treeset.vala  
  inflating: libgee-0.18.0/gee/unfolditerator.vala  
  inflating: libgee-0.18.0/gee/unrolledlinkedlist.vala  
  inflating: libgee-0.18.0/gitlog-to-changelog  
  inflating: libgee-0.18.0/libgee.doap  
   creating: libgee-0.18.0/m4/
  inflating: libgee-0.18.0/m4/vala.m4  
   creating: libgee-0.18.0/tests/
  inflating: libgee-0.18.0/tests/Makefile.am  
  inflating: libgee-0.18.0/tests/testarraylist.vala  
  inflating: libgee-0.18.0/tests/testarrayqueue.vala  
  inflating: libgee-0.18.0/tests/testbidirlist.vala  
  inflating: libgee-0.18.0/tests/testbidirsortedmap.vala  
  inflating: libgee-0.18.0/tests/testbidirsortedset.vala  
  inflating: libgee-0.18.0/tests/testcase.vala  
  inflating: libgee-0.18.0/tests/testcollection.vala  
  inflating: libgee-0.18.0/tests/testconcurrentlist.vala  
  inflating: libgee-0.18.0/tests/testconcurrentset.vala  
  inflating: libgee-0.18.0/tests/testdata.vala  
  inflating: libgee-0.18.0/tests/testdeque.vala  
  inflating: libgee-0.18.0/tests/testfunctions.vala  
  inflating: libgee-0.18.0/tests/testhashmap.vala  
  inflating: libgee-0.18.0/tests/testhashmultimap.vala  
  inflating: libgee-0.18.0/tests/testhashmultiset.vala  
  inflating: libgee-0.18.0/tests/testhashset.vala  
  inflating: libgee-0.18.0/tests/testlinkedlist.vala  
  inflating: libgee-0.18.0/tests/testlinkedlistasdeque.vala  
  inflating: libgee-0.18.0/tests/testlist.vala  
  inflating: libgee-0.18.0/tests/testmain.vala  
  inflating: libgee-0.18.0/tests/testmap.vala  
  inflating: libgee-0.18.0/tests/testmultimap.vala  
  inflating: libgee-0.18.0/tests/testmultiset.vala  
  inflating: libgee-0.18.0/tests/testpriorityqueue.vala  
  inflating: libgee-0.18.0/tests/testqueue.vala  
  inflating: libgee-0.18.0/tests/testreadonlybidirlist.vala  
  inflating: libgee-0.18.0/tests/testreadonlycollection.vala  
  inflating: libgee-0.18.0/tests/testreadonlylist.vala  
  inflating: libgee-0.18.0/tests/testreadonlymap.vala  
  inflating: libgee-0.18.0/tests/testreadonlyset.vala  
  inflating: libgee-0.18.0/tests/testset.vala  
  inflating: libgee-0.18.0/tests/testsortedmap.vala  
  inflating: libgee-0.18.0/tests/testsortedset.vala  
  inflating: libgee-0.18.0/tests/testtreemap.vala  
  inflating: libgee-0.18.0/tests/testtreemultimap.vala  
  inflating: libgee-0.18.0/tests/testtreemultiset.vala  
  inflating: libgee-0.18.0/tests/testtreeset.vala  
  inflating: libgee-0.18.0/tests/testunrolledlinkedlist.vala  
  inflating: libgee-0.18.0/tests/testunrolledlinkedlistasdeque.vala  
   creating: libgee-0.18.0/utils/
  inflating: libgee-0.18.0/utils/assume.h  
  inflating: libgee-0.18.0/utils/async.h  
  inflating: libgee-0.18.0/utils/free.h  
  inflating: libgee-0.18.0/utils/geeutils.vapi  
  inflating: libgee-0.18.0/utils/misc.h  
lightmapfuture.vala:44.2-44.23: error: overriding method `Gee.LightMapFuture.wait_until' is incompatible with base method `Gee.Future.wait_until': incompatible type of parameter 2.
abstractmultimap.vala:182.3-182.26: error: overriding method `Gee.AbstractMultiMap.Values.add' is incompatible with base method `Gee.AbstractCollection.add': incompatible type of parameter 1.
abstractmultimap.vala:182.3-182.26: error: Gee.AbstractMultiMap.Values.add: no suitable method found to override
abstractmultimap.vala:186.3-186.30: error: overriding method `Gee.AbstractMultiMap.Values.remove' is incompatible with base method `Gee.AbstractCollection.remove': incompatible type of parameter 1.
abstractmultimap.vala:186.3-186.30: error: Gee.AbstractMultiMap.Values.remove: no suitable method found to override
Failed: 5 error(s), 0 warning(s)
```

**Comment by btkostner** [comment]
I'm going to attach this to an upstream issue about how the `libgee` documentation is generated. For more information / progress look here: https://bugzilla.gnome.org/show_bug.cgi?id=774457

**Comment by astavale** [comment]
Thanks for reporting back. From the log files it looks as though valadoc-org should switch to using libgee-0.18.1 instead of 0.18.0. 

There have been some recent improvements to Vala's type checking for generics and the "incompatible type of parameter" errors suggest these two fixes are needed for libgee:
https://git.gnome.org/browse/libgee/commit/?id=a3a3cd81a3c69438e7a6219619b21fbf2b891ba1
https://git.gnome.org/browse/libgee/commit/?id=606cbe41ae8a3d7188807e79733224705c9837f7
Both of these fixes are present in 0.18.1.
